sycl/doc/extensions/supported/sycl_ext_oneapi_dot_accumulate.asciidoc

```diff
@@ -1,74 +1,99 @@
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
 
-Copyright (c) 2020 Intel Corporation.  All rights reserved.
 
 == Status
 
-Working Draft
+This extension is implemented and fully supported by {dpcpp}.
 
-This is a preview extension specification, intended to provide early access to a feature for review and community feedback. When the feature matures, this specification may be released as a formal extension.
 
-Because the interfaces defined by this specification are not final and are subject to change they are not intended to be used by shipping software products.
+== Overview
 
-== Version
+This extension adds new SYCL built-in functions that may simplify development
+and provide access specialized hardware instructions when a SYCL kernel needs
+to perform a dot product of two vectors followed by a scalar accumulation.
 
-Built On: {docdate} +
-Revision: 3
 
-== Contact
+== Specification
 
-Ben Ashbaugh, Intel (ben 'dot' ashbaugh 'at' intel 'dot' com)
+=== Feature test macro
 
-== Dependencies
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_DOT_ACCUMULATE` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
 
-This extension is written against the SYCL 1.2.1 specification, Revision v1.2.1-6.
+[%header,cols="1,5"]
+|===
+|Value
+|Description
 
-== Overview
+|1
+|Initial version of this extension.
+|===
 
-This extension adds new SYCL built-in functions that may simplify development and provide access specialized hardware instructions when a SYCL kernel needs to perform a dot product of two vectors followed by a scalar accumulation.
+=== New functions to compute the dot product of vectors
 
-== Enabling the extension
+This extension adds the following free functions:
 
-The extension is always enabled. The dot product functionality may be emulated in software or executed using hardware when suitable instructions are available.
+[source,c++]
+----
+namespace sycl::ext::oneapi {
 
-== Modifications of SYCL 1.2.1 specification
+int32_t dot_acc(vec<int8_t,4>  a, vec<int8_t,4>  b, int32_t c);
+int32_t dot_acc(vec<int8_t,4>  a, vec<uint8_t,4> b, int32_t c);
+int32_t dot_acc(vec<uint8_t,4> a, vec<int8_t,4>  b, int32_t c);
+int32_t dot_acc(vec<uint8_t,4> a, vec<uint8_t,4> b, int32_t c);
 
-=== Add to Section 4.13.6 - Geometric Functions
+int32_t doc_acc_4x8packed(int32_t a, int32_t b, int32_t c);
+int32_t doc_acc_4x8packed(int32_t a, uint32_t b, int32_t c);
+int32_t doc_acc_4x8packed(uint32_t a, int32_t b, int32_t c);
+int32_t doc_acc_4x8packed(uint32_t a, uint32_t b, int32_t c);
```

Comment:
I don't have a terribly strong opinion about either of these, but noting for completeness: If we want to align even closer to the related OpenCL extension we should consider the following changes:

1. The OpenCL extension only has `dot_acc` functions that do the dot product and accumulate for the variants that saturate, otherwise the function is simply `dot`, which does a dot product.
2. In addition to the return value difference, the type for the packed `a` and `b` arguments is unconditionally an unsigned integer in the OpenCL extension.  To disambiguate the different types the signedness of the argument is encoded in the function name (the return type is also), so for example if the packed `a` is signed and `b` is unsigned the full function is `int dot_4x8packed_su_int(uint a, uint b)`.

Links for reference:
* OpenCL C: https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_Ext.html#cl_khr_integer_dot_product
* SPIR-V: http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/KHR/SPV_KHR_integer_dot_product.html

Reply:
I like the suggestion to use `uint32_t` always for the "packed" parameters, so I also adopted the "_ss" (etc.) convention.  Done in 84f1ff887f8c5cc725ea3248330dd52b5844ea82.

It didn't make sense to me to adopt the final "_int" / "_uint" signifying the return type for the packed APIs unless we also adopt that convention for the unpacked APIs.  I decided to leave it off for both.

I don't know why this extension added `dot_acc` rather than just `dot`.  Dropping the "accumulate" operation seemed like a bigger change, so I decided not to do that here.